sqlmesh/core/macros.py

```diff
@@ -222,7 +231,20 @@ def template(self, text: t.Any, local_variables: t.Dict[str, t.Any]) -> str:
         Returns:
            The rendered string.
         """
-        return MacroStrTemplate(str(text)).safe_substitute(self.locals, **local_variables)
+        mapping = {}
+
+        for k, v in chain(self.locals.items(), local_variables.items()):
+            if not isinstance(v, str):
+                try:
+                    v = exp.convert(v)
```

Comment:
I'm not sure I follow what's going on in this block. An explanatory comment would be appreciated. Like why do we convert but ignore the failure? Is there a better way to know whether we should convert or not?

Reply:
i’d have to enumerate every type. additionally there’s a recursive error due to magic mock

Reply:
+1 about using a comment here